internal/controllers/topology/cluster/patches/variables/value.go

```diff
@@ -31,6 +31,27 @@ const (
 	rightArrayDelim = "]"
 )
 
+// NotFoundError is used when a variable is not found.
+type NotFoundError struct {
+	err error
+}
+
+// AsNotFoundError transforms the passed error into a NotFoundError.
+func AsNotFoundError(err error) NotFoundError {
```

Comment:
I'm not sure if there is another pattern we should use to create the error but could we unexport this func?

(I know it's just an internal package, but I would prefer also only exposing what is necessary here)